#3578 Add the ability to add a data attribute to <HiddenSelect /> to handle false a11y positives (#4759)
adobe/react-spectrum: packages/@react-aria/select/docs/useSelect.mdx

In right-to-left languages, the select should be mirrored. The arrow should be on the left,
and the selected value should be on the right. In addition, the content of list options should
flip. Ensure that your CSS accounts for this.

### False positive on `<HiddenSelect />` in accessibility tools

When using &lt;<TypeLink links={docs.links} type={docs.exports.HiddenSelect} />&gt; some accessibility tools such as AXE or pa11y might give a false positive.

> The data attribute `data-rsp-a11y-ignore` is included on the hidden select element and can be used to ignore the false positives in these tools.

Reviewer: For the team, open to opinions on where this information should go. Perhaps we should add a new section in this docs page for accessibility gotchas (or simply accessiblity?)

Reviewer 2: I'm fine with where it is, but not against a dedicated accessibility section either.

Reviewer 3: +1 to Reid's comment. I think this is fine where it is but a new section for accessibility gotchas would be nice

Reviewer: Alright, I'll approve this PR and make a followup one moving it to a separate accessibility section.

Reviewer 4: It definitely shouldn't be under "Internationalization" IMO, seems unrelated to that. Also we will need to cover this on all of the other select pages: RAC Select and RSP Picker. If it is going to be a general way for us to indicate false positives, we may also want to document it on a more general page such as the "Accessibility" one.

Reviewer: good point about covering this on the other Select pages. By the "Accessibility" page I assume you are referencing [this](https://react-spectrum.adobe.com/react-aria/accessibility.html#accessibility)? If so, I think that is a good idea.

@abitbetterthanyesterday IMO this is creeping in scope a bit, would you like to take this on? I'd be perfectly happy making followup PR for this as well instead.

> The data attribute `data-rsp-a11y-ignore` is included on the hidden select element and can be used to ignore the false positives in these tools.

Reviewer 5: ```suggestion
The data attribute `data-a11y-ignore` is included on the hidden select element and can be used to ignore the false positives in these tools.
```
It got renamed right?

Reviewer: I'll update in a followup

> The option `hasA11yIgnoreDataAttr` exposes a data attribute `data-rsp-a11y-ignore` on the hidden select element that can be used to ignore the false positives.

Reviewer 2: ```suggestion
The data attribute `data-rsp-a11y-ignore` is included on the hidden select element and can be used to ignore the false positives in these tools.
```

Reviewer 2: Hope its ok, went ahead and committed this change since we are looking to get this merged 🚀
